cabby/geo/visualize.py

```diff
@@ -0,0 +1,116 @@
+# limitations under the License.
+
+'''Library to support geographical visualization.'''
+
+from typing import Tuple, Sequence, Optional, Dict, Text
+from json import JSONDecoder, JSONDecodeError
+import json
+import pandas as pd
+import folium
+import geopandas as gpd
+import shapely.geometry as geom
+from shapely.geometry import Polygon, Point, LineString
+
+
+from cabby.geo import util
+
+def read_file(path: Text) -> gpd.GeoDataFrame:
+    '''Read file.
+    Arguments:
+      path: path to file.
+    Returns:
+      The GeoDataFrame theat includes the start and end points, pivots and 
+      route.
+    '''
+
+    start = gpd.read_file(path, layer='start')
+    end = gpd.read_file(path, layer='end')
+    route = gpd.read_file(path, layer='route')
+    main = gpd.read_file(path, layer='main')
+    near = gpd.read_file(path, layer='near')
+
+    start = start.rename(columns={'geometry': 'start_point'})
+    start['end_point'] = end['geometry']
+    start['route'] = route['geometry']
+    start['main_pivot_point'] = main['geometry']
+    start['near_pivot_point'] = near['geometry']
+
+    return start
+
+
+def get_osm_map(gdf: gpd.GeoDataFrame) -> Sequence[folium.Map]:
+    '''Create the OSM maps.
+    Arguments:
+      gdf: the GeoDataFrame from which to create the OSM map.
+    Returns:
+      OSM maps from the GeoDataFrame.
+    '''
+
+    mid_point = util.midpoint(gdf.end_point, gdf.start_point)
+    zoom_location = util.list_yx_from_point(mid_point)
+
+    # create a map
+    map_osm = folium.Map(location=zoom_location,
+                         zoom_start=15, tiles='OpenStreetMap')
+
+    # draw the points
+    start_point = util.list_yx_from_point(gdf.start_point)
+    end_point = util.list_yx_from_point(gdf.end_point)
+    main_pivot = util.list_yx_from_point(gdf.main_pivot_point)
+    near_pivot = util.list_yx_from_point(gdf.near_pivot_point)
+
+    folium.Marker(start_point, popup='start: ' + gdf.start, \
+      icon=folium.Icon(color='pink')).add_to(map_osm)
+    folium.Marker(end_point, popup='end: ' + gdf.end, \
+      icon=folium.Icon(color='black')).add_to(map_osm)
+    folium.Marker(main_pivot, popup='main pivot: ' + \
+      gdf.main_pivot, icon=folium.Icon(
+        color='orange', icon='info-sign')).add_to(map_osm)
+    folium.Marker(near_pivot, popup='near pivot: '+gdf.near_pivot, \
+       icon=folium.Icon(
+        color='red', icon='info-sign')).add_to(map_osm)
+
+    place_lng, place_lat = gdf.route.exterior.xy
+
+    points = []
+    for i in range(len(place_lat)):
+        points.append([place_lat[i], place_lng[i]])
+
+    for index, lat in enumerate(place_lat):
+        folium.Circle(location = [lat,
+                       place_lng[index]],
+                       radius = 5,
+                      color='crimson',
+
+                      ).add_to(map_osm)
+
+    return map_osm
+
+
+def get_maps_and_instructions(path: Text) -> Tuple[Sequence[folium.Map], \
+  Sequence[Text]]:
```

Comment:
Would be better to break up the line like this:
```
def get_maps_and_instructions(
  path: Text) -> Tuple[Sequence[folium.Map], Sequence[Text]]:
```

Reply:
Changed